src/components/multi-select/multi-select.jsx

```diff
@@ -0,0 +1,342 @@
+  const [autocompleteValue, setAutocompleteValue] = useState('');
+  const autocompleteRef = useRef();
+  const backupRef = useRef();
+  const [expanded, setExpanded] = useState(false);
+  const selfRef = ref || backupRef;
+  const [validationMessage, setValidationMessage] = useState(props.validationMessage);
+  const [value, setValue] = useState(props.value || []);
+  const valueRefs = value.map(() => createRef());
+  const visibleOptions = getVisibleOptions();
+  const visibleOptionsRefs = visibleOptions.map(() => createRef());
+  const wrapperRef = useRef(null);
+
+  const ids = {
+    emptyMessage: `${props.id}-empty`,
+    label: `${props.id}-label`,
+    listbox: `${props.id}-listbox`,
+    textbox: `${props.id}-autocomplete`,
+  };
+
+  const classNames = {
+    container: styles.container,
+    formControlChildrenContainer: getFormControlChildrenContainerClassName(props.readOnly, validationMessage),
+    iconClear: styles['icon-clear'],
+    iconsContainer: styles['icon-container'],
+    input: props.readOnly ? styles['input-readonly'] : styles.input,
+    noOptionsContainer: styles['no-options'],
+    popupContainer: styles['popup-container'],
+    tagList: styles['tag-list'],
+    ...props.classNames,
+  };
+
+  const dropdownContents = () => {
+    if (!expanded) {
+      return undefined;
+    }
+
+    if (props.showLoader) {
+      return (
+        <Listbox
+          className={classNames.popupContainer}
+          id={ids.listbox}
+          labelledBy={ids.label}
+          refs={[]}
+        >
+          <Loader inline />
+        </Listbox>
+      );
+    }
+
+    if (visibleOptions.length === 0) {
+      return (<NoOptions className={classNames.noOptionsContainer} id={ids.emptyMessage} />);
+    }
+
+    return (
+      <Listbox
+        className={classNames.popupContainer}
+        id={ids.listbox}
+        labelledBy={ids.label}
+        refs={visibleOptionsRefs}
+      >
+        {props.listboxChildren ?
+          props.listboxChildren(visibleOptions, visibleOptionsRefs, onOptionSelect) :
+          visibleOptions.map((option, index) => (
+            <ListOption
+              key={`${props.id}-${props.optionIDGetter(option)}`}
+              onSelect={onOptionSelect}
+              ref={visibleOptionsRefs[index]}
+              value={props.optionIDGetter(option)}
+            >
+              {props.optionLabelGetter(option)}
+            </ListOption>
+          ))}
+      </Listbox>
+    );
+  };
+
+  function getOption(id) {
+    return props.options.find(option => props.optionIDGetter(option) === id);
+  }
+
+  function getVisibleOptions() {
+    return props.options
+      .filter(option => props.optionLabelGetter(option).toLowerCase().includes(autocompleteValue.toLowerCase()))
+      .filter(option => !value.some((val) => {
+        const optionID = props.optionIDGetter(option);
+
+        if (typeof optionID === 'string') {
+          return val.toLowerCase() === optionID.toLowerCase();
```

Comment:
I would wager that the fact that these are not the same before lower casing implies they are different. 

If you happen to have two option tags,

- `<option value="test-1">test 1</option>`
- `<option value="Test-1">Test 1</option>`

Then you would assume those two are unique, different options, correct?

Reply:
This code was removed, and hopefully the intent clarified now that we don't support string literal options. For posterity, this was detecting options as string literals and providing case-insensitive filtering with autocompleter input for that case.